### PR TITLE
chore(ECO-265): Make Commonly Used Actions and Workflows Reusable

### PR DESCRIPTION
**Pull request by jpshilton-op**
- allows composite actions to be run in subdirs for client and types
- updates exiting testing workflows to use new setup action (which give us consistent caching)
- adds some missing parameters to existing workflows (b1367d4)
- fixes publishing to get token from secrets (68f76dd)
- fixes script to run 'tests' on client and types (860386c)
- consolidates setting of node version to one action

### CODE REVIEW

File: .github/workflows/test-integration-postgres-redis-rabbit.yml [hunk 1]
```diff
@@ -17,6 +17,11 @@ on:
         default: true
         required: false
         type: boolean
+      global_postgres_ref:
+        description: 'Branch of Postgres @openphone/database repo to use'
+        default: 'master'
+        required: false
+        type: string
```
Comment (Wagan8r) [inline .github/workflows/test-integration-postgres-redis-rabbit.yml:24]: Was this also added by mistake? I don't see it referenced in the workflow.

Comment (Wagan8r) [reply .github/workflows/test-integration-postgres-redis-rabbit.yml:24]: Ah, nvm. I had to load more of the file cause GH only gives the diffs.

Comment (jpshilton-op) [reply .github/workflows/test-integration-postgres-redis-rabbit.yml:24]: Yes both inputs are referenced below

File: .github/workflows/test-integration-elastic.yml [hunk 2]
```diff
@@ -22,10 +22,20 @@ on:
         default: 'http://localhost:9200'
         required: false
         type: string
+      postgres_host:
+        description: 'The Postgres host'
+        default: 'localhost'
+        required: false
+        type: string
```
[the remark this answers is not in the document]
Comment (jpshilton-op) [reply .github/workflows/test-integration-elastic.yml]: It was added because it is referenced below. I have removed it completely. c54d4ee

**Review by Wagan8r — APPROVED** [review]

**Pull request merged**